### PR TITLE
User service

### PR DESCRIPTION
**Pull request by moustaphastar**
1. Implemented new endpoints and objects in order
- to get a paged list of users
- to get a single user
- delete a user
2. Added access rule for delete and list endpoints
3. Implemented user page, details modal and api request for admin panel.

### CODE REVIEW

**Comment by moustaphastar** [comment]
Create user (#31 ) implementation is missing.

File: petshop-admin-panel/src/containers/user/User.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import CssBaseline from "@material-ui/core/CssBaseline";
+import Container from "@material-ui/core/Container";
+import CustomTable from "../../components/CustomTable";
+import {pageUsers, getUserById, deleteUserById} from "./UserService";
+import {createConfirmAlert, createErrorAlert, createSuccessAlert} from "../../components/Alert";
+import ViewUserModal from "./modal/ViewUserModal";
+
+
+const columns = [
+    "id", "Ad", "Email", "Telefon", "Lokasyon", "İşlemler"
```
Comment (moustaphastar) [inline petshop-admin-panel/src/containers/user/User.js:11]: What columns to show/hide?

File: petshop-admin-panel/src/containers/user/UserService.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+export const addUserImage= data => {
+    const formData = new FormData();
+    formData.append('file', data);
+
+    const addUserImageRequest = {
+        method: "POST",
+        path: "/api/v1/file",
+        data: formData,
+        params: ["firebasePath"],
+        values: ["product-images"]
```
Comment (moustaphastar) [inline petshop-admin-panel/src/containers/user/UserService.js:62]: Value should be `user-image`

Comment (moustaphastar) [reply petshop-admin-panel/src/containers/user/UserService.js:62]: Also, do we have the implementation to add a user image?

File: petshop-be/service/src/main/java/com/turkninja/petshop/user/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -47,7 +62,56 @@ public void signUp(UserSignupRequest request) {
...
+        UserEntity entity = userRepository.findByIdAndActiveTrue(id).orElseThrow(() ->
+                new ApplicationException(AppMessage.RECORD_NOT_FOUND,
+                        AppParameter.get("userId", id)));
+        entity.setActive(false);
+        userRepository.save(entity);
+    }
 
+    @Override
+    public UserImageInsertResponse insertUserImage(UserImageInsertRequest request) {
+        return new UserImageInsertResponse();
```
Comment (moustaphastar) [inline petshop-be/service/src/main/java/com/turkninja/petshop/user/impl/UserServiceImpl.java:113]: User image insertion is done through `FileResource` endpoints.

File: petshop-be/web/src/main/java/com/turkninja/petshop/v1/UserResource.java [hunk 4]
```diff
@@ -26,4 +35,40 @@ public UserResource(UserService userService) {
         userService.signUp(request);
         return new ResponseEntity<>(HttpStatus.CREATED);
     }
+
+    @GetMapping()
+    public ResponseEntity<PageResponse<UserResponse>> pageList(
+            @RequestParam("page") int page,
+            @RequestParam("size") int size) {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
```
Comment (moustaphastar) [inline petshop-be/web/src/main/java/com/turkninja/petshop/v1/UserResource.java:43]: Can we refactor to use `hasRole` or `hasAuthority` annotations?

**Review by moustaphastar — COMMENTED** [review]
CreateUser (#31) missing.

File: petshop-be/domain/src/main/java/com/turkninja/petshop/mapper/UserMapper.java [hunk 5]
```diff
@@ -1,14 +1,61 @@
...
+    @Mapping(target = "town",
+            expression = "java(entity.getAddress().getTown())")
+    @Mapping(target = "lastLoggedIn",
+            dateFormat = "dd-MM-yyyy HH:mm:ss")
+    @Mapping(target = "createdAt",
+            dateFormat = "dd-MM-yyyy HH:mm:ss")
+    @Mapping(target = "revisedAt",
+            dateFormat = "dd-MM-yyyy HH:mm:ss")
+    UserResponse entityToUserResponse(UserEntity entity);
+
```
Comment (aliturgutbozkurt) [inline petshop-be/domain/src/main/java/com/turkninja/petshop/mapper/UserMapper.java:52]: We don't need target attribute for entity properties that has same name with response property

**Review by aliturgutbozkurt — APPROVED** [review]

**Review by moustaphastar — COMMENTED** [review]
Reviewed.

**Pull request merged**